src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1998,13 +2015,29 @@ private void HandleVariableDeclaration(IVariableDeclarationOperation operation)
                     }
                 }
 
+                // If we have a previousBlock, then we must have an initializer to ensure we don't create empty regions that can't be cleaned up.
+                Debug.Assert(previousBlock == null || initializer != null);
+
                 if (initializer != null)
                 {
                     // We can't use the IdentifierToken as the syntax for the local reference, so we use the
                     // entire declarator as the node
                     var localRef = new LocalReferenceExpression(localSymbol, isDeclaration: true, semanticModel: null, declarator.Syntax, localSymbol.Type, constantValue: default, isImplicit: true);
                     var assignment = new SimpleAssignmentExpression(localRef, isRef: localSymbol.IsRef, initializer, semanticModel: null, assignmentSyntax, localRef.Type, constantValue: default, isImplicit: true);
                     AddStatement(assignment);
+
+                    if (localSymbol.IsStatic)
+                    {
+                        BasicBlock initializerBlock = CurrentBasicBlock;
+
+                        var initializationSemaphore = new StaticLocalInitializationSemaphoreOperation(localSymbol, localSymbol.DeclaringSyntaxReferences.Single().GetSyntax(), type: null, constantValue: null, isImplicit: true);
+                        Operation.SetParentOperation(initializationSemaphore, null);
+
+                        _currentBasicBlock = null;
+                        LeaveRegion();
+                        LinkBlocks(previousBlock, CurrentBasicBlock);
+                        LinkBlocks(previousBlock, (initializationSemaphore, JumpIfTrue: true, RegularBranch(initializerBlock)));
```

Comment:
>RegularBranch(initializerBlock) [](start = 94, length = 31)

This doesn't look right. This links to the very last block in the initializer, when we should be linking to the first one.
Also, I suggest to move all code dealing with the Semaphor "business" right before we enter initialization region. Allocate new block (afterInitialization) there and link it with the CurrentBasicBlock as appropriate, enter the region and link the conditional branch. Here, leave the region and add the afterInitialization block.